packages/core/core/src/requests/AssetRequestRunner.js

```diff
@@ -122,7 +118,8 @@ export default class AssetRequestRunner extends RequestRunner<
       for (let [moduleSpecifier, version] of result.devDeps) {
         let depVersionRequst = {
           moduleSpecifier,
-          resolveFrom: result.resolvedPath, // TODO: resolveFrom should be nearest package boundary
+          resolveFrom:
+            result.pkgFilePath != null ? result.pkgFilePath : result.searchPath,
```

Comment:
Not entirely sure how this should be handled. Should we enforce the pkgFilePath to be defined?

Would be rather slow to enforce resolving on each config. Seems to work fine for now though and it wasn't implemented properly before either as it had a `TODO` next to it...

Reply:
If we change `addDevDependency` to require `resolveFrom` as the second argument then we'll use that.